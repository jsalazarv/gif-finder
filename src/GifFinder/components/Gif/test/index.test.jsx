import { render, screen } from '@testing-library/react';
import { Gif } from '../index';

describe('[Gif]', () => {
  describe('[Snapshot]', () => {
    test('Should render [Gif] component and contain class [gif]', () => {
      const { container } = render(<Gif />);

      expect(container.firstChild).toHaveClass('gif');
      expect(container).toMatchSnapshot();
    });

    test('Should contain the text that is received as props', () => {
      const title = 'Test title';
      const { container } = render(<Gif title={title} />);

      expect(container).toHaveTextContent('Test title');
    });

    test('Should contain the URL that is received as props', () => {
      const url = 'https://url-test.com/';
      render(<Gif url={url} />);
      const anchor = screen.getByRole('anchor');

      expect(anchor).toHaveProperty('href', 'https://url-test.com/');
    });
  });
});
