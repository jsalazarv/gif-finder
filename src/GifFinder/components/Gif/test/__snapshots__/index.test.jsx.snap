// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Gif] [Snapshot] Should render [Gif] component and contain class [gif] 1`] = `
<div>
  <div
    class="gif"
  >
    <div
      class="image-container"
    >
      <img />
    </div>
    <a
      role="anchor"
      target="_blank"
    >
      <small />
    </a>
  </div>
</div>
`;
